hyperspy/tests/signal/test_signal_operators.py

```diff
@@ -17,154 +17,94 @@
-        assert_true((n.data ==
-                     self.s1.data + self.s2.data).all())
-        for i, size in enumerate(n.data.shape):
-            assert_equal(size, n.axes_manager._axes[i].size)
-
-    def test_s2_plus_s1(self):
-        n = self.s2 + self.s1
-        assert_true((n.data ==
-                     self.s1.data + self.s2.data).all())
-        for i, size in enumerate(n.data.shape):
-            assert_equal(size, n.axes_manager._axes[i].size)
-
-    def test_s3_plus_s1(self):
-        n = self.s3 + self.s1
-        assert_true((n.data ==
-                     self.s1.data + self.s2.data).all())
-        for i, size in enumerate(n.data.shape):
-            assert_equal(size, n.axes_manager._axes[i].size)
-
-
-class TestBinaryOperatorsCase3:
-
-    """The signals are not aligned but can be aligned because their
-    shapes are as follows:
-
-    +---------+------------------+-------------+
-    | Signal  | NavigationShape  | SignalShape |
-    +---------+------------------+-------------+
-    +---------+------------------+-------------+
-    |   s1    |       (0,)       |      a      |
-    +---------+------------------+-------------+
-    |   s2    |       (0,)       |      b      |
-    +---------+------------------+-------------+
-
-    """
-
-    def setUp(self):
-        self.s1 = Signal(np.arange(10).reshape(2, 5))
-        self.s1.axes_manager.set_signal_dimension(2)
-        self.s2 = Signal(np.arange(6))
-
-    def test_s1_times_s2(self):
-        n = self.s1 * self.s2
-        assert_true((n.data ==
-                     self.s1.data[np.newaxis, ...] *
-                     self.s2.data[:, np.newaxis, np.newaxis]).all())
-        for i, size in enumerate(n.data.shape):
-            assert_equal(size, n.axes_manager._axes[i].size)
-        assert_equal(n.axes_manager.signal_dimension, 2)
-
-    def test_s2_times_s1(self):
-        n = self.s2 * self.s1
-        assert_true((n.data ==
-                     (self.s2.data[np.newaxis, np.newaxis, :] *
-                      self.s1.data[..., np.newaxis])).all())
-        for i, size in enumerate(n.data.shape):
-            assert_equal(size, n.axes_manager._axes[i].size)
-        assert_equal(n.axes_manager.signal_dimension, 1)
-
-
-class TestBinaryOperatorsCase4:
-
-    """Signal and number.
-
-    """
-
-    def setUp(self):
-        self.s1 = Signal(np.arange(6))
-
-    def test_plus_right(self):
-        assert_true(((self.s1 + 2).data == self.s1.data + 2).all())
-
-    @raises(TypeError)
-    def test_left_right(self):
-        assert_true(((2 + self.s1).data == self.s1.data + 2).all())
+        self.s1 = Signal(np.ones((2, 3)))
+        self.s2 = Signal(np.ones((2, 3)))
+        self.s2.data *= 2
+
+    def test_sum_same_shape_signals(self):
+        s = self.s1 + self.s2
+        assert_array_equal(s.data, self.s1.data * 3)
+
+    def test_sum_in_place_same_shape_signals(self):
+        self.s1 += self.s2
+        assert_array_equal(self.s1.data, np.ones((2, 3)) * 3)
```

Comment:
Should check that the `id(self.s1)` is the same before and after (to make sure the operation is actually performed in-place).

Other places lower down as well